### PR TITLE
Refactor StringIO/ByteIO changes made to is_match functions

### PR DESCRIPTION
**Pull request by ChrisWallace2020**
Mostly PEP8 compliance and cleaning up some logic for readability and conciseness.

### CODE REVIEW

File: dataprofiler/data_readers/filepath_or_buffer.py [hunk 1]
```diff
@@ -31,35 +33,43 @@ def __init__(self, filepath_or_buffer, open_method='r', encoding=None, seek_offs
...
 
         if self.seek_offset is not None:
             self._filepath_or_buffer.seek(self.seek_offset, self.seek_whence)
 
         return self._filepath_or_buffer
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO, DetachingTextIOWrapper)):
+        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO,
+                                                 DetachingTextIOWrapper)):
```
Comment (JGSweets) [inline dataprofiler/data_readers/filepath_or_buffer.py:58]: If we instead on `__init__` stored a variable `self._is_wrapped` could we get rid of `DetachingTextIOWrapper`?

Basically we could check that variable to see if we need to detach.prior to this if. If so, then do a swap so to speak.
```python
wrapper = self._filepath_or_buffer
self._filepath_or_buffer = wrapper.buffer
wrapper.detach()
```

File: dataprofiler/data_readers/filepath_or_buffer.py [hunk 2]
```diff
@@ -27,39 +29,39 @@ def __init__(self, filepath_or_buffer, open_method='r', encoding=None, seek_offs
...
 
         return self._filepath_or_buffer
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO, DetachingTextIOWrapper)):
+        # Need to detach buffer if wrapped (i.e. BytesIO opened with 'r'
+        if self._is_wrapped:
+            wrapper = self._filepath_or_buffer
+            self._filepath_or_buffer = wrapper.buffer
+            wrapper.detach()
```
Comment (ChrisWallace2020) [inline dataprofiler/data_readers/filepath_or_buffer.py:62]: This is the functionality needed from `DetachingTextIOWrapper`, that is now streamlined and clearer

**Review by JGSweets — APPROVED** [review]

**Review by AnhTruong — APPROVED** [review]
